### PR TITLE
Telegram function

### PR DESCRIPTION
**Pull request by kongzii**
Screenshot by Dropbox Capture

### CODE REVIEW

File: prediction_market_agent/agents/microchain_agent/call_api.py [hunk 1]
```diff
@@ -57,4 +53,27 @@ def __call__(
...
+    ) -> str:
+        url = f"https://api.telegram.org/bot{APIKeys().telegram_bot_key.get_secret_value()}/sendMessage?chat_id={chat_id}&text={message}"
+        response = requests.get(url)
+        response.raise_for_status()
+        return response.text
+
+
+API_FUNCTIONS: list[type[Function]] = [
+    CallAPI,
+    SendTelegramMessage,
```
Comment (evangriffiths) [inline prediction_market_agent/agents/microchain_agent/call_api.py:78]: Feels odd for the trader agent to have this in its functions by default. I've made this new issue to address this https://github.com/gnosis/prediction-market-agent/issues/350. How do you feel about not having this always registered by the trader/just-born agents until we solve that?

File: prediction_market_agent/agents/microchain_agent/call_api.py [hunk 2]
```diff
@@ -57,4 +53,27 @@ def __call__(
...
+
+    def __call__(
+        self,
+        chat_id: str,
+        message: str,
+    ) -> str:
+        url = f"https://api.telegram.org/bot{APIKeys().telegram_bot_key.get_secret_value()}/sendMessage?chat_id={chat_id}&text={message}"
+        response = requests.get(url)
+        response.raise_for_status()
+        return response.text
```
[the remark this answers is not in the document]
Comment (kongzii) [reply prediction_market_agent/agents/microchain_agent/call_api.py:73]: As in the screenshot, not very helpful, changed.

File: prediction_market_agent/agents/microchain_agent/call_api.py (hunk 1 above)
Comment (kongzii) [reply prediction_market_agent/agents/microchain_agent/call_api.py]: It's not needed atm so no feelings at all 😄  Commented out

File: prediction_market_agent/agents/microchain_agent/call_api.py [hunk 3]
```diff
@@ -54,7 +50,30 @@ def __call__(
             headers=json.loads(headers) if headers else None,
         )
         response.raise_for_status()
+        return "Message sent"
```
[the remark this answers is not in the document]
Comment (evangriffiths) [reply prediction_market_agent/agents/microchain_agent/call_api.py]: @coderabbitai I think you're missing some context. See https://github.com/gnosis/prediction-market-agent/pull/349#discussion_r1698378043

**Review by evangriffiths — APPROVED** [review]

**Pull request merged**